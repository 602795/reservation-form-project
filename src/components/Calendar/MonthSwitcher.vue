<template>
  <div class="month-switcher d-flex flex-column justify-center">
    <div class="content d-flex justify-around align-center">
      <span @click="$emit('prev-month')" class="mdi mdi-chevron-left icon-size"></span>
      <div class="font-bold">{{ month }} {{ year }}</div>
      <span @click="$emit('next-month')" class="mdi mdi-chevron-right icon-size"></span>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  components: {},
})
export default class MonthSwitcher extends Vue {
  @Prop()
  readonly month!: string;

  @Prop()
  readonly year!: string;
}
</script>

<style lang="scss">
  .month-switcher {
    border: 1px solid #D8D8D8;
    border-radius: 100px;
    height: var(--chips-height);
    min-height: var(--chips-height);

    .content {
      line-height: 20px;
    }
  }
</style>
