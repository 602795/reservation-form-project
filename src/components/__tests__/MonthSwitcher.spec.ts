import { mount, Wrapper } from '@vue/test-utils';
import MonthSwitcher from '@/components/ReservationForm/Calendar/MonthSwitcher.vue';
import { Vue } from 'vue/types/vue';
import moment from 'Moment';

describe('Month Switcher Component Test', () => {
  let wrapper: Wrapper<Vue>;

  beforeEach(() => {
    wrapper = mount(MonthSwitcher, {
      propsData: {
        baseMoment: moment().clone(),
      },
      mocks: {
        $moment: moment,
      }
    });
  });

  test('Emituje event po kliknięciu w poprzedni miesiąc',  () => {
    const icon = wrapper.find('.month-switcher .content .mdi-chevron-right');
    icon.trigger('click');

    expect(wrapper.emitted('next-month')?.length).toBe(1);
  });

  test('Nie możemy kliknąć w poprzedni miesiąc',  () => {
    const icon = wrapper.find('.month-switcher .content .mdi-chevron-left');
    icon.trigger('click');

    expect(wrapper.emitted('prev-month')?.length).toBe(undefined);
  });
});