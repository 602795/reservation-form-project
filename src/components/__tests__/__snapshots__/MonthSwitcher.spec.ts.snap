// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month Switcher Component Test Pasuje do snapchota 1`] = `
<div class="month-switcher d-flex flex-column justify-center">
  <div class="content d-flex justify-around align-center"><span class="mdi mdi-chevron-left icon-size disabled"></span>
    <div class="font-bold"> </div> <span class="mdi mdi-chevron-right icon-size"></span>
  </div>
</div>
`;
